### PR TITLE
ParameterizeSQL

### PR DESCRIPTION
**Pull request by jannyHou**
connect to strongloop-internal/scrum-apex#232
### Checklist
- [x] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style
  guide](http://loopback.io/doc/en/contrib/style-guide.html)

### CODE REVIEW

File: test/dashdb.injection.test.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+          {name: 'John', pass: '456'},
+      ];
+      Customer.create(samples, done);
+    });
+  });
+  it('does not return result with SQL injection', function(done) {
+    // generate injectedSQL as `SELECT * FROM Cusotmer WHERE name ="" or ""=""`
+    // ""="" is always true
+    var injectedFilter = {
+      where: {or: [{name: ''}, {'': ''}]},
```
Comment (kjdelisle) [inline test/dashdb.injection.test.js]: Can we change the injection to assert something like the prototypical `1=1`?

Comment (kjdelisle) [inline test/dashdb.injection.test.js]: If it's the same in other connectors we test for injection, then that's fine.

Comment (jannyHou) [inline test/dashdb.injection.test.js]: sure, I find both of them in some SQL injection tutorial, I mean `1=1` and `""=""`

File: lib/dashdb.js [hunk 2]
```diff
@@ -113,9 +124,9 @@ DASHDB.prototype.createTable = function(model, cb) {
                         'currently supported')));
   } else {
     tasks.push(function(callback) {
-      var sql = 'CREATE TABLE ' + tableSchema + '.' + tableName +
-          ' (' + columnDefinitions + ') ORGANIZE BY ROW;';
-      self.execute(sql, callback);
+      var sql = new ParameterizedSQL('CREATE TABLE ' + tableSchema + '.' +
+        tableName + ' (' + columnDefinitions + ') ORGANIZE BY ROW;');
+      self.execute(sql.sql, callback);
```
Comment (b-admike) [inline lib/dashdb.js]: If you're calling the base connector's `execute` function here, I think you can do `self.execute(sql.sql, sql.params, opts, callback)` here.

File: test/dashdb.injection.test.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+      if (err) return done(err);
+      var samples = [
+          {name: 'Zoe', pass: '123'},
+          {name: 'John', pass: '456'},
+      ];
+      Customer.create(samples, done);
+    });
+  });
+  it('does not return result with SQL injection', function(done) {
+    // generate injectedSQL as `SELECT * FROM Cusotmer WHERE name ="" or ""=""`
```
Comment (b-admike) [inline test/dashdb.injection.test.js]: Typeo `Cusotmer` -> `Customer` and if you changed the filter to be `1=1`, perhaps update the comment as well

Comment (jannyHou) [inline test/dashdb.injection.test.js]: lol thanks, actually I updated it on local and about to squash commits then push it with dependency change(if needed.) together.

File: lib/dashdb.js (hunk 2 above)
Comment (jannyHou) [inline lib/dashdb.js]: good catch, changed.

Comment (jannyHou) [inline lib/dashdb.js]: there is no `options` in this function, so omit `opts` here.

**Pull request merged**